### PR TITLE
Add basic custom Translation Tracker syntax

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- Changes to the docs site code

#### Description

This PR adds a rudimentary version of the custom tracker syntax to help with the 3.0 PRs changing lots of translations to fix links (LOOKING AT YOU 3.0 IMAGE GUIDE).

This allows you to add a `@tracker-major:(list of file paths separated by ;)` to a final squashed commit description (the one you have to confirm before you merge) that makes all changes be considered minor except the ones you have explicitly marked as major.

🚨 **THE FILE PATH HAS TO INCLUDE EVERYTHING (SEE BELOW)** 🚨 

Example:

**PR updates the pages (br/1.mdx, br/2.mdx), but also updates the link to another page that isn't fully updated yet (br/3.mdx).**
Using: `@tracker-major:./src/content/docs/br/1.mdx;./src/content/docs/br/2.mdx`, `br/3.mdx` continues to be marked as outdated.

More info in the code comment!

**How to test it?**
1. Pull the branch
2. Run `pnpm translation-status` (or with your preferred package manager) 
3. Open the `dist/translation-status/index.html` file. (this is the current status in the branch!)
4. Now, make a new commit changing any pages you want, and add the custom syntax to the commit description.
5. Do steps 2 & 3 again.
6. See the changes done in the tracker. Might be helpful to have `i18n.docs.astro.build` side by side, but will be inconsistent the moment the branch isn't 100% updated in relation to main.

### CODE REVIEW

File: scripts/lib/translation-status/builder.ts [hunk 1]
```diff
@@ -325,6 +326,38 @@ export class TranslationStatusBuilder {
...
+		
+	*/
+	isValidMajor(entry: DefaultLogFields & ListLogLine, filePath: string) {
+		if (entry.message.match(COMMIT_IGNORE)) return false;
+
+		const trackerDirective = entry.body.match(TRACKER_DIRECTIVE);
+
+		if (trackerDirective) {
+			const filePaths = trackerDirective[0].replace('@tracker-major:', '').split(';');
+			return filePaths.indexOf(filePath) > -1;
```
Comment (dreyfus92) [inline scripts/lib/translation-status/builder.ts:355]:
```suggestion
			return filePaths.includes(filePath);
```

File: scripts/lib/translation-status/builder.ts [hunk 2]
```diff
@@ -325,6 +326,38 @@ export class TranslationStatusBuilder {
...
+		to these pages all become outdated and other pages from the commit stay the same. 
+		Also works when there's no English pages! The translated files will be marked as 
+		updated, but the same pages from other languages won't be affected by it, keeping
+		their old state.
+		
+	*/
+	isValidMajor(entry: DefaultLogFields & ListLogLine, filePath: string) {
+		if (entry.message.match(COMMIT_IGNORE)) return false;
+
+		const trackerDirective = entry.body.match(TRACKER_DIRECTIVE);
```
Comment (dreyfus92) [inline scripts/lib/translation-status/builder.ts:351]:
```suggestion
		const trackerDirectiveMatch = entry.body.match(TRACKER_DIRECTIVE);
```

File: scripts/lib/translation-status/builder.ts [hunk 3]
```diff
@@ -325,6 +326,38 @@ export class TranslationStatusBuilder {
...
+
+		if (trackerDirective) {
+			const filePaths = trackerDirective[0].replace('@tracker-major:', '').split(';');
```
Comment (dreyfus92) [inline scripts/lib/translation-status/builder.ts:354]:
```suggestion
			const filePaths = trackerDirectiveMatch[0].replace('@tracker-major:', '').split(';');
```

**Review by dreyfus92 — CHANGES_REQUESTED** [review]
Awesome job @Yan-Thomas. I left you a couple of minor suggestions to improve a bit code readibility. Let me know what you think 😁

File: scripts/lib/translation-status/builder.ts [hunk 4]
```diff
@@ -325,6 +326,38 @@ export class TranslationStatusBuilder {
...
+		updated, but the same pages from other languages won't be affected by it, keeping
+		their old state.
+		
+	*/
+	isValidMajor(entry: DefaultLogFields & ListLogLine, filePath: string) {
+		if (entry.message.match(COMMIT_IGNORE)) return false;
+
+		const trackerDirectiveMatch = entry.body.match(TRACKER_DIRECTIVE);
+
+		if (trackerDirective) {
```
Comment (dreyfus92) [inline scripts/lib/translation-status/builder.ts:353]:
```suggestion
		if (trackerDirectiveMatch) {
```

Comment (dreyfus92) [reply scripts/lib/translation-status/builder.ts:353]: This might fix the error logged by the CI / Check

**Review by TheOtterlord — APPROVED** [review]
Tested locally :partying_face:

**Comment by Yan-Thomas** [comment]
Thanks peeps! Let's get this merged 🚀

**Pull request merged**